### PR TITLE
Docs: Mention implicit TLS port requirements for SMTP

### PR DESCRIPTION
**Pull request by chenbt-hz**
Added additional description for smtp configuration.

This is helpful for using the grafana email alert function. 

Especially for developers using smtp proxy servers. 

**To solve this problem, it may be necessary to reconsider the security policy of smtp connection, or to add a new configuration switch.Adding this descriptive information may avoid many enterprise users from encountering the same doubts.**

### CODE REVIEW

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:1099]:
```suggestion
Default is `localhost:25`. Port 465 is only supported for the SMTP proxy server with the TLS port enabled.
```

**Comment by chri2547** [comment]
@grafana/user-essentials , could I request a review of this external contributor PR? I've made some wording suggestions, but want to also get Engr signoff. Thank you.

**Comment by chri2547** [comment]
> @grafana/backend-platform , wondering if I could get a quick review of this external PR? I've made wording suggestions, but would like Engr signoff.

> ^^ Pinging again, please. @grafana/backend-platform , wondering if I could get a quick review of this external PR? I've made wording suggestions, but would like Engr signoff.

**Comment by chri2547** [comment]
@grafana/grafana-frontend-platform , can you please take a look at this PR and approve if it looks ok to you. Thank you!

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (sakjur) [inline docs/sources/setup-grafana/configure-grafana/_index.md:1099]: SMTP proxy server doesn't seem like the right wording here, and since the entire section is for configuring an SMTP mail transfer agent of some kind I think we can omit that part entirely.

I'm also struggling a bit with how to formulate this to be correct, since STARTTLS is supported on other ports as well. In lieu of a better wording, I believe we can borrow [RFC8314](https://www.rfc-editor.org/rfc/rfc8314)'s "Implicit TLS" here, since that provides a searchable term that is fairly well-used and less prone to misunderstanding than "SMTPS" which isn't really a thing and is described as several different things around the web.

It's also a bit unclear whether this means that port 465 is only supported for TLS (yes) or TLS is only supported on port 465 (also yes). My suggestion doesn't make that any clearer, but I think it's clearer _that_ it doesn't make that clearer with the terser slightly more instructive rather than descriptive sentence structure. If that makes any sense 😅

@chri2547 WDYT?

```suggestion
Default is `localhost:25`. Use port 465 for implicit TLS.
```

Comment (chenbt-hz) [reply docs/sources/setup-grafana/configure-grafana/_index.md:1099]: Thank you for your reply~
I originally meant TLS is only supported on port 465.
For the code:
// NewDialer returns a new SMTP Dialer. The given parameters are used to connect
// to the SMTP server.
func NewDialer(host string, port int, username, password string) *Dialer {
	return &Dialer{
		Host:         host,
		Port:         port,
		Username:     username,
		Password:     password,
		SSL:          port == 465,                    _**<--- here !!!**_
		Timeout:      10 * time.Second,
		RetryFailure: true,
	}
But my English proficiency is not enough for me to be able to chew words. I believe you can find a more suitable description to avoid other users encountering the same problem.

**Review by chri2547 — APPROVED** [review]

**Comment by sakjur** [comment]
@chenbt-hz Thank you for the PR 👍

**Pull request merged**
